app/controllers/proposals_controller.rb

```diff
@@ -0,0 +1,62 @@
+class ProposalsController < ApplicationController
+  before_action :load_pq, only: [:new, :create]
+  before_action :authenticate_user, only: [:new, :create]
+
+  def new
+    @action_officers = ActionOfficer.active_list
+  end
+
+  def create
+    proposal_form = ProposalForm.new(create_params)
+    if proposal_form.valid?
+      pq = ProposalService.new.propose(proposal_form)
+      flash[:success] = 'Successfully proposed Deputy Director(s)'
+      redirect_to early_bird_landing_page_path
+    else
+      flash[:error] = 'Please choose a Deputy Director.'
+      redirect_to new_pq_proposal_path(@pq.id)
+    end
+  end
+
+  private
+
+  def authenticate_user
+    redirect_to_early_bird_dashboard unless current_user || user_has_valid_early_bird_credentials?
+  end
+
+  def redirect_to_early_bird_dashboard
+    token = session[:early_bird_token]
```

Comment:
Should this method use `early_bird_landing_page_path` in case the session variables don't exist?

Reply:
If the session variable doesn't exist then the early_bird_landing_page will be unaccessible.